src/bin/cfsctl.rs

```diff
@@ -179,25 +251,88 @@ async fn main() -> Result<()> {
             OciCommand::MetaLayer { ref name } => {
                 oci::meta_layer(&repo, name, None)?;
             }
-            OciCommand::PrepareBoot { ref name, bootdir } => {
-                let output = bootdir.unwrap_or(PathBuf::from("/boot"));
-                oci::prepare_boot(&repo, name, None, &output)?;
+            OciCommand::PrepareBoot {
+                ref config_name,
+                ref config_verity,
+                ref bootdir,
+                ref entry_id,
+                ref cmdline,
+            } => {
+                let verity = verity_opt(config_verity)?;
+                let mut fs = oci::image::create_filesystem(&repo, config_name, verity.as_ref())?;
+                let entries = fs.transform_for_boot(&repo)?;
+                let id = fs.commit_image(&repo, None)?;
+
+                let Some(entry) = entries.into_iter().next() else {
```

Comment:
We discussed this in a live chat, my preference here is that when we create a sealed image we also by default prune the old `vmlinuz+initramfs.img`. Yes, the RPM database would say it's missing, but I don't see that as a serious problem. We already have other related issues with moving `/var/lib/selinux` back to `/etc`.

The advantage of pruning is we don't duplicate the kernel binary, and it's very clear - there's one kernel in the image.

In theory of course we could have the official `prepare-boot` flow run through `rpmbuild` + `rpm -Uvh` (for fedora-bootc derivatives); actually really on this topic it's important to note there's already `kernel-uki-virt` and so we could suggest installing `kernel-uki-stub` or something that we then mutate?

Reply:
My 2¢: we should modify the image as little as possible.

If the person creating the image wants to delete the files then they absolutely can.  If the files are in the image that we've already downloaded them and have them in the repository.   Having them in the image is no extra overhead.